pandas/core/categorical.py

```diff
@@ -1364,6 +1365,54 @@ def sort_values(self, inplace=False, ascending=True, na_position='last'):
             return self._constructor(values=codes, categories=self.categories,
                                      ordered=self.ordered, fastpath=True)
 
+    def rank(self, method='average', na_option='keep',
+             ascending=True, pct=False):
```

Comment:
Other proposal: we make this a private function (`_rank`), and than the full docstring is not needed (just some explanation of why this function is here and what it does).

IMO it is not needed to have a public `Categorical.rank`